lib/arproxy/proxy_chain.rb

```diff
@@ -35,13 +35,28 @@ def execute_with_arproxy(sql, name=nil, **kwargs)
         end
         alias_method :execute_without_arproxy, :execute
         alias_method :execute, :execute_with_arproxy
+
+        if private_method_defined?(:raw_execute)
+          private
+          def raw_execute_with_arproxy(sql, name, **kwargs)
```

Comment:
```suggestion
          private def raw_execute_with_arproxy(sql, name, **kwargs)
```

changing further method definition to `private` conditionally poses a risk to cause a bug!